### PR TITLE
Add missing oauth params

### PR DESCRIPTION
**Pull request by jer-tx**

### CODE REVIEW

File: pyramid_oauthlib/__init__.py [hunk 1]
```diff
@@ -18,9 +19,11 @@
     'password',
     'refresh_token',
     'response_type',
+    'response_mode',
```
Comment (tilgovi) [inline pyramid_oauthlib/__init__.py:22]: It looks like this is part of OIDC. Maybe we can, just for readability and ease of comparison against oauthlib, separate this into parts with a comment, like here: https://github.com/oauthlib/oauthlib/blob/v2.1.0/oauthlib/common.py#L396

**Comment by tilgovi** [comment]
Looking at the base `Request` class from oauthlib, I'm not sure it's 100% correct to add `request.client` here. The request validator may use a custom sub-class of `Client`, for instance, that can load and persist its client_secret from a DB model.

What's sort of surprising is that the documentation [recommends](https://oauthlib.readthedocs.io/en/v2.1.0/oauth2/validator.html#oauthlib.oauth2.RequestValidator.authenticate_client_id) setting `request.client` in the `authenticate_client` or `authenticate_client_id` methods of the request validator, but the error you posted in #18 seems to suggest that it's required.

I think what's going on is that oauthlib is flexible, almost to a fault, here. If you were to implement custom grant types, they might not need `request.client`, but the built-in ones do (or at least, authorization_code does). Therefore, it's incumbent upon you to set `request.client` to something useful to you (which may subclass `Client`) if your grant types rely on it.

**Comment by tilgovi** [comment]
I think this is where examples / documentation would help a lot.

So, for now, maybe we can add all the OIDC parameters (and the missing OAuth 2 parameter) and just merge that.

**Comment by jer-tx** [comment]
> So, for now, maybe we can add all the OIDC parameters (and the missing OAuth 2 parameter) and just merge that.

I can add all the OIDC params, which oauth2 param are you referring to?

**Comment by tilgovi** [comment]
I was referring to `user`, which you added. And I think, yeah, we should add all those OIDC params while we're here. Thank you!

File: pyramid_oauthlib/__init__.py [hunk 2]
```diff
@@ -21,7 +21,20 @@
     'redirect_uri',
     'scope',
     'state',
+    'user'
```
Comment (tilgovi) [inline pyramid_oauthlib/__init__.py:24]: I think we might be missing `token` and `token_type_hint` here, still.

Comment (tilgovi) [reply pyramid_oauthlib/__init__.py:24]: It looks like they also have `scopes`. So many! Sorry!

Comment (tilgovi) [reply pyramid_oauthlib/__init__.py:24]: Yikes, and `redirect_uri`. Maybe we just copy from here: https://github.com/oauthlib/oauthlib/blob/v2.1.0/oauthlib/common.py#L396

😆

Comment (jer-tx) [reply pyramid_oauthlib/__init__.py]: Done and done

**Comment by tilgovi** [comment]
Actually, before I merge this (since it's been a little while), can you confirm for me that it's still possible to set `request.client`, overriding the getter that `add_oauth_param` installs if we list `client` in this set?

**Comment by jer-tx** [comment]
As in, can we still set `request.client` on the validator if `client` exists in `OAUTH_PARAMS`? yes that works. If I add `config.add_oauth_param('client')` to config, I get a `pyramid.exceptions.ConfigurationConflictError` error, which is probably intended behavior.

**Comment by tilgovi** [comment]
Thanks, I've been in JS-land so much lately I think I was thinking of `Object.defineProperty()` making non-configurable properties (can't be reassigned/deleted). By adding a request property in Pyramid is not that invasive, so we're good.

**Pull request merged**